### DISCUSSION TITLE
Who can help me to retrieve the IPv6 address of the operating system using V?

### DISCUSSION THREAD

**Discussion by Jengro777** [vlang/v#18983] [Questions and Answers] [answered]
Who can help me to retrieve the IPv6 address of the operating system using V?

**Comment by Jengro777** [comment]
Can rewards for answers be set here? The adopted answer can receive a cash reward, which will be paid by the questioner.

**Reply by JalonSolov** [reply]
V supporters usually answer for free, if they know the answer.

**Comment by Wajinn** [comment, marked as answer]
It might be hard for people to understand what is exactly wanted by the question.  Different OSes have different ways to show the IPv6 address.  On Windows (example 1) you can use `ipconfig` (under Link-local IPv6 Address) or `netsh` .  If you are looking for the IPv6 address of other sites, you could use `resolve_addrs` from the `net` module.

This is assuming that IPv6 was setup on the computer, because it's not always the case.  Oh, and there are lots of different examples for doing things in V on Exercism and Rosetta Code, which has become quite helpful.

Example 1:
```
import  os

fn main() {
//	result := os.execute('netsh interface ipv6 show address')
	result := os.execute('ipconfig /all')	
	if result.output !='' {println(result.output)}
	else {println('Error: not working') exit(1)}
}
```
Example 2 (from V's Rosetta Code DNS query example):
```
import net

fn main() {
	addr := 'www.kame.net:80'
	@type := net.SocketType.tcp
	family := net.AddrFamily.unspec
	mut addrs := []net.Addr{}
	mut results :=''

	addrs = net.resolve_addrs(addr, family, @type) or {println('Error: nothing resolved') exit(1)}
	for each in addrs {
		results += '${addr.split(':')[0]} * ${each} * ${each.family()} * ${@type} \n'
	}
	println(results)
}
```
